pkg/controller/psbackup/controller_test.go

```diff
@@ -406,6 +407,116 @@ func TestRunningState(t *testing.T) {
 	}
 }
 
+func TestGetBackupSource(t *testing.T) {
+	scheme := runtime.NewScheme()
+	if err := clientgoscheme.AddToScheme(scheme); err != nil {
+		t.Fatal(err, "failed to add client-go scheme")
+	}
+	if err := apiv1alpha1.AddToScheme(scheme); err != nil {
+		t.Fatal(err, "failed to add apis scheme")
+	}
+
+	ctx := context.Background()
+
+	tests := []struct {
+		name    string
+		cr      *apiv1alpha1.PerconaServerMySQLBackup
+		cluster *apiv1alpha1.PerconaServerMySQL
+		want    string
+		wantErr bool
+	}{
+		{
+			name: "sourceHost from backup",
+			cr: &apiv1alpha1.PerconaServerMySQLBackup{
+				Spec: apiv1alpha1.PerconaServerMySQLBackupSpec{
+					SourceHost: "backuphost",
+				},
+			},
+			cluster: &apiv1alpha1.PerconaServerMySQL{
+				Spec: apiv1alpha1.PerconaServerMySQLSpec{
+					Backup: &apiv1alpha1.BackupSpec{SourceHost: "clusterhost"},
+				},
+			},
+			want:    "backuphost",
+			wantErr: false,
+		},
+		{
+			name: "host from cluster",
+			cr: &apiv1alpha1.PerconaServerMySQLBackup{
+				Spec: apiv1alpha1.PerconaServerMySQLBackupSpec{},
+			},
+			cluster: &apiv1alpha1.PerconaServerMySQL{
+				Spec: apiv1alpha1.PerconaServerMySQLSpec{
+					Backup: &apiv1alpha1.BackupSpec{SourceHost: "clusterhost"},
+				},
+			},
+			want:    "clusterhost",
+			wantErr: false,
+		},
+		{
+			name: "single node cluster",
+			cr:   &apiv1alpha1.PerconaServerMySQLBackup{},
+			cluster: &apiv1alpha1.PerconaServerMySQL{
+				Spec: apiv1alpha1.PerconaServerMySQLSpec{
+					MySQL: apiv1alpha1.MySQLSpec{
+						PodSpec: apiv1alpha1.PodSpec{Size: 1},
+					},
+					Backup: &apiv1alpha1.BackupSpec{},
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "single-node",
+					Namespace: "test-ns",
+				},
+			},
+			want:    "single-node-mysql-0.single-node-mysql.test-ns",
+			wantErr: false,
+		},
+		{
+			name: "async cluster, orchestrator off, no host",
+			cr:   &apiv1alpha1.PerconaServerMySQLBackup{},
+			cluster: &apiv1alpha1.PerconaServerMySQL{
+				Spec: apiv1alpha1.PerconaServerMySQLSpec{
+					MySQL: apiv1alpha1.MySQLSpec{
+						ClusterType: apiv1alpha1.ClusterTypeAsync,
+					},
+					Backup: &apiv1alpha1.BackupSpec{},
+					Unsafe: apiv1alpha1.UnsafeFlags{
+						Orchestrator: true,
+					},
+					Orchestrator: apiv1alpha1.OrchestratorSpec{
+						Enabled: false,
+					},
+				},
+			},
+			want:    "",
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		cb := fake.NewClientBuilder().WithScheme(scheme).WithObjects(tt.cr).WithStatusSubresource(tt.cr)
+		if tt.cluster != nil {
+			cb.WithObjects(tt.cluster)
+		}
+
+		r := PerconaServerMySQLBackupReconciler{
+			Client:        cb.Build(),
+			Scheme:        scheme,
+			ServerVersion: &platform.ServerVersion{Platform: platform.PlatformKubernetes},
+		}
+
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := r.getBackupSource(ctx, tt.cr, tt.cluster)
+			if tt.wantErr {
+				assert.Error(t, err)
```

Comment:
Instead of simply checking if an error exists, we can check more specifically that the error we want is throw i,e, `Orchestrator is disabled, ...`